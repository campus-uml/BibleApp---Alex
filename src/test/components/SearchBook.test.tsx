import { render, screen, fireEvent } from "@testing-library/react";
import { describe, test, expect } from "vitest";
import SearchBook from "../../components/SearchBook";
import { vi } from "vitest";

describe("SearchBook Component", () => {
test("renderiza el campo de entrada con el marcador de posiciÃ³n", () => {
    render(<SearchBook onSearch={() => {}} />);
    const input = screen.getByPlaceholderText("Buscar en la Biblia...");
    expect(input).toBeInTheDocument();
});

test("llama a onSearch al escribir en el input", () => {
    const onSearchMock = vi.fn();
    render(<SearchBook onSearch={onSearchMock} />);
    const input = screen.getByPlaceholderText("Buscar en la Biblia...");
    
    fireEvent.change(input, { target: { value: "Genesis" } });
    expect(onSearchMock).toHaveBeenCalledTimes(1);
    expect(onSearchMock).toHaveBeenCalledWith("Genesis");
  });
});
