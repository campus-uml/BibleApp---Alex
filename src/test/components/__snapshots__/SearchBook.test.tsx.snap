// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SearchBook Component > matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="relative"
  >
    <svg
      class="lucide lucide-search absolute left-2 top-1/2 h-4 w-4 -translate-y-1/2 transform text-muted-foreground"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="11"
        cy="11"
        r="8"
      />
      <path
        d="m21 21-4.3-4.3"
      />
    </svg>
    <input
      class="flex h-9 w-full rounded-md border border-input bg-transparent px-3 py-1 text-base shadow-sm transition-colors file:border-0 file:bg-transparent file:text-sm file:font-medium file:text-foreground placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:cursor-not-allowed disabled:opacity-50 md:text-sm pl-8"
      placeholder="Buscar en la Biblia..."
      type="search"
    />
  </div>
</DocumentFragment>
`;
